CMakeLists.txt

```diff
@@ -108,9 +108,11 @@ function(importLibraries)
     "Linux:librpm"
     "Linux:libudev"
     "Linux,Darwin,Windows:libxml2"
+    "Darwin,Windows:lief"
```

Comment:
Why not include linux?

Reply:
This work doesn't add Linux support, also we already have tables parsing the ELF, but using another library (libelfin).
As @puffyCid was mentioning, we were thinking of using the same library for all three platforms, but the work for the ELF ones is a conversion one.

I would suggest maybe doing that work in another PR, since this one has already quite a bit going on.

Reply:
I may be reading this wrong, but if lief is cross platform, can we _also_ support the lief based tables cross platform?

Reply:
as @Smjert  mentioned, right now linux is using a different library to parse ELF files. I think it would be great to move the parsing under LIEF but im not sure if that should be included in this PR or a separate one. It looks like maybe a separate PR is preferred?

Reply:
If we thought this was eventually going to be multiplatform, I'd generally bias towards doing that work cleanly and upfront. Even if we didn't move the table generation.

But I don't have that strong a feeling, and I'm not the one doing the work. So kinda up to y'all

Reply:
After further comments, it was preferred to migrate the `elf tables` now rather than later.  
All the `elf tables` should now be using LIEF.  I tried to make tables identical to the originals and i think they are ~90% identical. There are a few small changes, but since the next release 5.0 is a breaking change release hopefully the changes will be allowed?
Small comparison:  
new table:
```
+---------+----------+---------+---------+-------------+---------+---------+---------+-------+--------+--------+------------+
| path    | filename | class   | abi     | abi_version | type    | machine | version | flags | is_pie | has_nx | entrypoint |
+---------+----------+---------+---------+-------------+---------+---------+---------+-------+--------+--------+------------+
| /bin/ls | ls       | CLASS64 | SYSTEMV | 0           | DYNAMIC | x86_64  | CURRENT | 0     | 1      | 1      | 67d0       |
+---------+----------+---------+---------+-------------+---------+---------+---------+-------+--------+--------+------------+
```
old table
```
osquery> select * from elf_info where path = '/bin/ls';
+-------+------+-------------+------+---------+---------+-------+-------+---------+
| class | abi  | abi_version | type | machine | version | entry | flags | path    |
+-------+------+-------------+------+---------+---------+-------+-------+---------+
| 64    | sysv | 0           | dyn  | 62      | 1       | 26576 | 0     | /bin/ls |
+-------+------+-------------+------+---------+---------+-------+-------+---------+
```